src/lib.rs

```diff
@@ -99,42 +99,42 @@ pub struct RcMutexGuardian<T: 'static> {
 impl<T> Deref for ArcRwLockReadGuardian<T> {
     type Target = T;
     fn deref(&self) -> &Self::Target {
-        &*self.inner
+        self.inner.as_ref().unwrap()
```

Comment:
Let's make all of these
```suggestion
        self.inner.as_ref().expect("inner is only None in drop")
```